tests/Drupal/Commenting/FunctionCommentUnitTest.inc

```diff
@@ -318,10 +318,10 @@ function test26($var) {
 /**
  * Data types with spaces.
  *
- * @param string or int $x
- *   Data types should not contain spaces.
+ * @param array<int, \stdClass> $x
```

Comment:
Hm, can we only allow spaces after a comma? That would be more strict and should satisfy your array<> use cases.

Ideally we should still flag such broken type specifications like here.

Reply:
I dont think this can work as things like `@return (T is int ? static : array<static>)` are possible.

I think at best types should be any non whitespace character. in the case of `@param` only one whitespace before $paramName, and only newline in the case of `@return`